### PR TITLE
Memory management stuff added

### PR DESCRIPTION
**Pull request by GuanHuiLim**

### CODE REVIEW

File: Engine/src/Engine/Memory/StackAllocator.cpp [hunk 1]
```diff
@@ -45,7 +41,6 @@ void* StackAllocator::Alloc(const Size size, const U8 alignment) {
 	}
 
 	m_top = newTop;
-
 	return reinterpret_cast<void*>(alignedAddress);
```
Comment (spironan) [inline Engine/src/Engine/Memory/StackAllocator.cpp:44]: does it need to be reinterpret cast here?

File: Engine/src/Engine/Memory/MemoryManager.cpp [hunk 2]
```diff
@@ -5,30 +5,30 @@ const MemoryManager::Size MemoryManager::BYTE = 1;
...
+	m_persistentAllocator(2 * MemoryManager::GB),
+	//m_poolAlloc(),
+	m_total_size(2 * MemoryManager::GB)
+{
+	//ENGINE_ASSERT_MSG(instance != nullptr, "MemoryManager instance already exists");
+	if (instance != nullptr)
+	{
+		throw std::bad_alloc{};
+		std::cout << "MemoryManager instance already exists" << std::endl;
+		return;
```
Comment (spironan) [inline Engine/src/Engine/Memory/MemoryManager.cpp:22]: will this lines every be hit since you throw already?

**Review by spironan — CHANGES_REQUESTED** [review]
While overloading new was explained can't be within a namespace can everything else that makes of the memory manger be wrapped in namespace? also do answer some of the queries highlighted below!

File: Engine/src/Engine/Memory/StackAllocator.cpp (hunk 1 above)
Comment (GuanHuiLim) [reply Engine/src/Engine/Memory/StackAllocator.cpp]: Yes, as static cast may change the data

File: Engine/src/Engine/Memory/MemoryManager.cpp (hunk 2 above)
Comment (GuanHuiLim) [reply Engine/src/Engine/Memory/MemoryManager.cpp]: updated to print it out before the throw

**Review by spironan — APPROVED** [review]
all good merging in.

**Pull request merged**